### PR TITLE
Update NETCore.App dependencies for nca1.0 and nca.1.1

### LINKED ISSUE #2512: Microsoft.NETCore.App 2.0 doesn't list netcoreapp1.1 dependencies
## Steps to reproduce
Look at the dependencies listed in the NETCore.App 2.0 package:

https://www.nuget.org/packages/Microsoft.NETCore.App/2.0.0-preview1-002111-00
https://dotnet.myget.org/feed/dotnet-core/package/nuget/Microsoft.NETCore.App/2.0.0-preview2-25324-02

## Expected  behavior
It should list the dependencies for netcoreapp1.0, netcoreapp1.1, and netcoreapp2.0

## Actual behavior
It only lists the dependencies for netcoreapp1.0 and netcoreapp2.0

@weshaggard @ericstj @gkhanna79

### PR DESCRIPTION
**Pull request by ericstj**
Dependencies were missing for netcoreapp1.1 due to missing import.  Fixes #2512

Update dependencies to latest stable versions to give folks serviced packages.  Related https://github.com/dotnet/standard/issues/354.

/cc @eerhardt @weshaggard @gkhanna79

### CODE REVIEW

File: src/pkg/projects/Microsoft.NETCore.App/netcoreapp1.1.props [hunk 1]
```diff
@@ -28,6 +28,12 @@
     <NETCoreApp11Dependency Include="NETStandard.Library">
```
Comment (weshaggard) [inline src/pkg/projects/Microsoft.NETCore.App/netcoreapp1.1.props]: Why didn't you update the runtime and host packages for 1.1? They should be up to 1.1.2

Comment (ericstj) [inline src/pkg/projects/Microsoft.NETCore.App/netcoreapp1.1.props]: Thanks, will do.

Comment (weshaggard) [inline src/pkg/projects/Microsoft.NETCore.App/netcoreapp1.1.props]: I would suggest diffing the nuspect for the 1.1.2 package with your local one to make sure the deps are updated.

Comment (ericstj) [inline src/pkg/projects/Microsoft.NETCore.App/netcoreapp1.1.props]: Sure.  I also had to lift some dependencies that were seemingly missed from 1.0/1.1 core.app packages.

File: src/pkg/projects/Microsoft.NETCore.App/netcoreapp1.0.props [hunk 2]
```diff
@@ -13,10 +13,13 @@
...
     </NETCoreApp10Dependency>
     <NETCoreApp10Dependency Include="Microsoft.NETCore.DotNetHostPolicy">
-      <Version>1.0.1</Version>
+      <Version>1.0.5</Version>
     </NETCoreApp10Dependency>
     <NETCoreApp10Dependency Include="Microsoft.NETCore.Runtime.CoreCLR">
-      <Version>1.0.4</Version>
+      <Version>1.0.7</Version>
+    </NETCoreApp10Dependency>
+    <NETCoreApp10Dependency Include="Microsoft.NETCore.Targets">
```
Comment (eerhardt) [inline src/pkg/projects/Microsoft.NETCore.App/netcoreapp1.0.props]: I don't see this dependency on [https://www.nuget.org/packages/Microsoft.NETCore.App/1.0.5](https://www.nuget.org/packages/Microsoft.NETCore.App/1.0.5)
But I DO see a dependency for Microsoft.NETCore.Platforms (>= 1.0.2), which I don't see here.

Comment (eerhardt) [inline src/pkg/projects/Microsoft.NETCore.App/netcoreapp1.0.props]: Also, 1.0.5 has a dependency on runtime.native.System.Security.Cryptography (>= 4.0.1) now.

---
In reply to: [118324972](https://github.com/dotnet/core-setup/pull/2515#discussion_r118324972) [](ancestors = 118324972)

Comment (ericstj) [inline src/pkg/projects/Microsoft.NETCore.App/netcoreapp1.0.props]: I'm doing a full diff to find gaps.

**Comment by ericstj** [comment]
Ok, now the diff is clean between 1.1.2 package and this.  I'm doing some assets file inspection to make sure everything looks good there as well.

**Comment by ericstj** [comment]
Assets file diff looks good.  One thing that's concerning that I mentioned above is that there are packages which have been serviced but those serviced versions aren't being consumed by the shared framework:
- NETCoreApp1.0
  - System.Diagnostics.StackTrace
    - Included: 4.0.1
    - Latest serviced: 4.0.2
  - System.IO.Compression
    - Included: 4.1.0
    - Latest serviced: 4.1.1

I'm not going to fix those here because that'd create a mismatch and make those packages go app-local.  I'll file an issue for servicing.

**Pull request merged**